lib/mix/lib/mix/tasks/escript.install.ex

```diff
@@ -28,21 +39,52 @@ defmodule Mix.Tasks.Escript.Install do
 
   ## Command line options
 
-    * `--sha512` - checks the escript matches the given SHA-512 checksum
+    * `--sha512` - checks the escript matches the given SHA-512 checksum. Only
+      applies to installations via URL or local path.
 
     * `--force` - forces installation without a shell prompt; primarily
       intended for automation in build systems like make
 
+    * `--submodules` - fetch repository submodules before building escript from
+      git or github
+
+    * `--app` - specify a custom app name to be used for building the escript
+      from git, github, or hex
   """
 
   @behaviour Mix.Local.Installer
 
   @escript_file_mode 0o555 # only read and execute permissions
 
-  @switches [force: :boolean, sha512: :string]
+  @switches [force: :boolean, sha512: :string, submodules: :boolean, app: :string]
   @spec run(OptionParser.argv) :: boolean
   def run(argv) do
-    Mix.Local.Installer.install({__MODULE__, :escript}, argv, @switches)
+    {opts, args} = OptionParser.parse!(argv, strict: @switches)
+
+    case Mix.Local.Installer.parse_args(args, opts) do
```

Comment:
Ideally we build merge this case with the one here: https://github.com/antipax/elixir/blob/5aeff0add7be3b6906ce790dee73d1a758173b86/lib/mix/lib/mix/local/installer.ex#L32 :)